projects/plugins/boost/app/modules/cache/Page_Cache.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Page_Cache;
+
+use Automattic\Jetpack_Boost\Contracts\Is_Always_On;
+use Automattic\Jetpack_Boost\Contracts\Pluggable;
+
+class Page_Cache implements Pluggable, Is_Always_On {
+	/*
+	 * Sets up the advanced-cache.php file and if that works, adds the WP_CACHE
+	 * define to wp-config.php
+	 * These are used by WordPress to load the caching system before most of
+	 * WordPress is loaded.
+	 */
+	public function setup() {
+		if ( is_admin() && isset( $_GET['page'] ) && $_GET['page'] === 'jetpack-boost' ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			require_once ABSPATH . '/wp-admin/includes/file.php';
+			WP_Filesystem();
+			$result = $this->create_advanced_cache();
+			if ( $result === true ) {
+				$result = $this->add_wp_cache_define();
+				if ( $result !== true ) {
+					return $result;
+				}
+			} else {
+				return $result;
+			}
+
+			return true;
+		}
+	}
+
+	/*
+	 * Creates the advanced-cache.php file.
+	 *
+	 * Returns true if the files were setup correctly, or WP_Error if there was a problem.
+	 * @return bool|WP_Error
+	 */
+	private function create_advanced_cache() {
+		global $wp_filesystem;
+
+		$advanced_cache_filename = ABSPATH . 'wp-content/advanced-cache.php';
+		if ( $wp_filesystem->exists( $advanced_cache_filename ) ) {
+			$content = $wp_filesystem->get_contents( $advanced_cache_filename );
+			if ( strpos( $content, 'Boost Cache Plugin 0.1' ) !== false ) {
+				return true;
+			} else {
+				return new \WP_Error( 'advanced-cache.php exists but is not the correct file' );
+			}
+		} else {
+			$contents = '<?php
+// Boost Cache Plugin 0.1
+require_once( ABSPATH . \'/wp-content/plugins/boost/app/modules/cache/Boost_File_Cache.php\' );
+
+( new Automattic\Jetpack_Boost\Modules\Page_Cache\Boost_File_Cache() )->serve();
+';
+
+			$result = $wp_filesystem->put_contents( $advanced_cache_filename, $contents, FS_CHMOD_FILE );
+			if ( $result === false ) {
+				return new \WP_Error( 'Could not write to advanced-cache.php' );
+			}
+		}
+
+		return true;
+	}
+
+	/*
+	 * Adds the WP_CACHE define to wp-config.php
+	 */
+	private function add_wp_cache_define() {
+		global $wp_filesystem;
+
+		$content = $wp_filesystem->get_contents( ABSPATH . 'wp-config.php' );
+		if ( strpos( $content, 'define( \'WP_CACHE\', true );' ) !== false ) {
+			return; // WP_CACHE already added.
+		}
```

Comment:
This check requires that `WP_CACHE` has been added using our coding standards, with spaces inside the brackets - and that the value is `true`. This won't identify:

```
define('WP_CACHE',true);
define( 'WP_CACHE', false );
```

Perhaps a regexp is the way to go here?

Also it may hit a false positive if there is a commented-out `WP_CACHE` define in the file - but detecting that is getting ridiculous. Perhaps it should check `defined( 'WP_CACHE' )` to rule out commented out lines, and use a regexp to detect anything that looks like any kind of `WP_CACHE` setter?

e.g.:
```
$is_defined = preg_match( '#define\s*\(\s*(["'])WP_CACHE\1\s*,[^\)]+\)#', $content ) && defined( 'WP_CACHE' );
$is_truthy = WP_CACHE;
```

?

Reply:
Looking at this again - doesn't WP have some function to check these?

Reply:
> Looking at this again - doesn't WP have some function to check these?

Well, it has tools to add blocks of text to the .htaccess file, but not to wp-config.php, but ithemes security uses something similar to add their own blocks of code to wp-config.php, while messing up WP Super Cache's WP_CACHE define.
There's no special commands for modifying the config file.

Reply:
I think a regex is the way to go. I added one that includes the string true with ', ", or bare.